### PR TITLE
chore: tests for chain abstraction

### PR DESCRIPTION
**Pull request by chris13524**
- [x] Cover `/status` endpoint & `check_in`
- [x] Reduce faucet use frequency i.e. 10x the requested amount of gas
- [x] ~~Faucet for USDC: always pull into address 1, chain 1~~ moved to [separate issue](https://linear.app/reown/issue/RES-130/tests-auto-usdc-faucet)
- [ ] ~~Enable in CI~~

Resolves WK-359

Resolves RES-130

Additional test coverage captured in: https://linear.app/reown/issue/RES-128/yttrium-tests-other-test-scenariosnon-happy-path

### CODE REVIEW

File: crates/yttrium/src/chain_abstraction/api/mod.rs [hunk 1]
```diff
@@ -10,18 +10,39 @@ uniffi::custom_type!(Address, String, {
...
+
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[cfg_attr(feature = "uniffi", derive(uniffi_macros::Record))]
 #[serde(rename_all = "camelCase")]
 pub struct Transaction {
     pub from: Address,
     pub to: Address,
-    pub value: String,
-    pub gas: String,
-    pub gas_price: String,
```
Comment (llbartekll) [inline crates/yttrium/src/chain_abstraction/api/mod.rs:21]: this change affects swift bindings, so please notify me of changes like this too

File: crates/yttrium/src/chain_abstraction/client.rs [hunk 2]
```diff
@@ -67,4 +80,57 @@ impl Client {
...
+    ) -> Result<StatusResponseCompleted, WaitForSuccessError> {
+        self.wait_for_success_with_timeout(
+            orchestration_id,
+            check_in,
+            Duration::from_secs(30),
+        )
+        .await
+    }
+
+    pub async fn wait_for_success_with_timeout(
```
Comment (jakubuid) [inline crates/yttrium/src/chain_abstraction/client.rs:96]: Why do we need `wait_for_success` and `wait_for_success_with_timeout`?
Doesn't BC API timeout itself?

File: crates/yttrium/src/chain_abstraction/api/mod.rs (hunk 1 above)
Comment (chris13524) [reply crates/yttrium/src/chain_abstraction/api/mod.rs:21]: I think this should fail CI then?

File: crates/yttrium/src/chain_abstraction/client.rs (hunk 2 above)
Comment (chris13524) [reply crates/yttrium/src/chain_abstraction/client.rs:96]: This timeout is how long the client will poll the Blockchain API's `/status` endpoint

**Review by llbartekll — APPROVED** [review]

File: crates/yttrium/src/chain_abstraction/client.rs (hunk 2 above)
Comment (jakubuid) [reply crates/yttrium/src/chain_abstraction/client.rs:93]: I get it but why client should specify that number? client doesn't know it and API should time out itself

File: crates/yttrium/src/chain_abstraction/api/mod.rs (hunk 1 above)
Comment (llbartekll) [reply crates/yttrium/src/chain_abstraction/api/mod.rs:21]: agree but unfortunately i don't have tests for types decoding and not gonna add them as we are going to change ffi lib

File: crates/yttrium/src/chain_abstraction/client.rs (hunk 2 above)
Comment (chris13524) [reply crates/yttrium/src/chain_abstraction/client.rs:96]: I think we still need a timeout here to handle server issues where what you suggest isn't implemented or is implemented incorrectly.

[Slack conversation](https://reown-inc.slack.com/archives/C0816SK4877/p1732211619422449) for establishing a server-side timeout

File: crates/yttrium/src/chain_abstraction/api/mod.rs (hunk 1 above)
Comment (chris13524) [reply crates/yttrium/src/chain_abstraction/api/mod.rs:21]: I added some `git diff`s at the end of the CI, hopefully this will catch this

**Pull request merged**